transport/conn.go

```diff
@@ -289,7 +295,12 @@ func (c *Conn) init() error {
 	}
 }
 
-// TODO add conn Close, make sure go routines exit
+// Close closes connection and terminates reader and writer go rutines.
+func (c *Conn) Close() {
+	_ = c.conn.Close()
+	close(c.w.requestCh)
+	c.r.clearHandlers()
```

Comment:
This should be called drainHandlers, close should be added to clearHandlers renamed to drainHandlers